### ISSUE TITLE
508-defect-2 [COGNITION]: H1 should match or be included in the page title

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#19552]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Josh_

## Details

"Direct deposit information" as an `h2` may cause confusion as the user will most likely expect this to be an `h1` since:
- the breadcrumb suggests that the user is on the direct deposit page
- the secondary navigation suggests that the user is on the direct deposit page
- the `title` suggests that the user is on the direct deposit page
- the visually styling suggests that the `h2` "Direct deposit information" is actually an `h1`
-  "Profile" technically does not have its own page; should a user click on "Profile" in the breadcrumbs, they will be taken to the "Personal and contact information" page which violates [WCAG 2.4.4 link purpose](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html#:~:text=The%20text%20of%2C%20or%20associated,the%20document%20or%20web%20application) and sets a false expectation that a "main" profile page exists. 

This may have deeper consequences should this break habituation/consistency on how headings work throughout VA.gov. Solving this will also most likely require a larger conversation as the core of this issue lies in IA, content, and design decisions.

## Acceptance Criteria

- [ ] The page title either matches the `h1` of the page or includes part of it 
- [ ] More acceptance criteria may be written depending on the approach chosen to solve this issue

## Environment

* Operating System, including: any
* Browser: all
* Screenreading device: any
* Server destination: staging

## Steps to Recreate

1. Enter `https://staging.va.gov/profile/direct-deposit` in browser
2. [Using the chrome extension headingsMap](https://chrome.google.com/webstore/detail/headingsmap/flbjommegcjonpdmenkdiocclhjacmbi?hl=en), view the headings of the page
3. Confirm that the `h1` is "Profile" which does not match the page title "Direct Deposit"

## Possible Solutions

Accessibility and content specialists have ideated a couple of possible solutions, but no final solution can be suggested until a larger conversation is held with the original design team. Potential solutions include:

### Create a real profile page that is a parent to all the existing pages. 
The left subnavigation "Profile" could be removed and the `h2`s would be bumped up to `h1`s. There may not be enough useful information to justify creating a parent page to satisfy this solution though.

### Convert the "Personal & Contact Information" page into the real profile page
Since "my profile" links to personal and contact information, we could establish it as the real profile page and set it as the parent of the other pages both hierarchically and visually. Semantically this may not be accurate if it is meant more to be a sibling.

### Add "Profile:" to the start of all the `h2` strings and bump them up to `h1`s. 
For example, instead of "Personal & Contact Information," rewrite it as "Personal & Contact Information | Profile." "Profile" could then be removed from the breadcrumbs. This may be repetitive and verbose, particularly to screen reader users-- but it may better satisfy accessibility needs.

### Remove the breadcrumb, set the Veteran's name as the `h1,` and update `title`
If a Veteran arrives on the profile page via the global nav, they will land on the "Personal & Contact Information" page and will hear the page's `title` e.g. "Charde Hardin's Profile | Personal & Contact Information." If they are navigating within the profile itself, focus will be set to the `h2` e.g. "Personal & Contact Information."

<img width="767" alt="Screen Shot 2021-03-15 at 3 49 39 PM" src="https://user-images.githubusercontent.com/14154792/111213311-02ec3800-85a7-11eb-90d6-8f5e9247dadf.png">

## WCAG or Vendor Guidance (optional)

* [defect-level-2 guidance on titles](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)
* [Page Titled: Understanding SC 2.4.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-title.html)

## Screenshots or Trace Logs
<img width="1792" alt="Screen Shot 2021-02-05 at 11 44 47 AM" src="https://user-images.githubusercontent.com/14154792/107065699-14257600-67ab-11eb-9f00-5a1d8ea9e2f1.png">

**Comment by joshkimux** [comment]
https://user-images.githubusercontent.com/14154792/107071686-1986be80-67b3-11eb-8c02-3c5348127bce.mov

**Comment by joshkimux** [comment]
The IA gets more confusing on mobile.

<img width="323" alt="Screen Shot 2021-02-08 at 3 39 29 PM" src="https://user-images.githubusercontent.com/14154792/107278506-e434d780-6a23-11eb-9b33-2046cbf28c80.png">
<img width="321" alt="Screen Shot 2021-02-08 at 3 39 39 PM" src="https://user-images.githubusercontent.com/14154792/107278512-e4cd6e00-6a23-11eb-9457-6daeae9d069b.png">

**Comment by joshkimux** [comment]
@mnorthuis [I saw this thread on slack talking about the my profile headings](https://dsva.slack.com/archives/CQV81GBBM/p1597084900202000) and wanted to bring this conversation up again as I may be missing important context that you've already worked through with Jennifer!

**Comment by mnorthuis** [comment]
Oh yes, that was a conversation...I'm open to more talk around this one, as I don't think we landed on a perfect solution to balance headings, title tags, and breadrumbs.    Happy to discuss options!

**Comment by mshea0606** [comment]
@erikphansen @sporeboy  We are changing the side nav to "Direct Deposit Information"

**Comment by mshea0606** [comment]
@erikphansen this is listed as a must/launch blocker in Peggy's feedback

**Comment by erikphansen** [comment]
@mshea0606 I noticed you added the `508-defect-1` label to this because Peggy had flagged it as a launch blocker. @joshkimux can correct me if I'm wrong, but I am think you probably shouldn't be messing with the `508-defect` labels since those are really specific to a11y issues and an accessibility specialist should be managing those labels on the ticket. Also seems weird to have both `508-defect-1` and `508-defect-2` labels on the same ticket :)

**Comment by erikphansen** [comment]
In actually reading this ticket, it sounds like the issues are far wider than simply changing the `Direct deposit` label to `Direct deposit information`. But those issues do not have a clear solution or path forward. So I submitted a PR to fix the menu, section, and document titles and will wait for future guidance on the other issues. We should leave it to @joshkimux to close this ticket after https://github.com/department-of-veterans-affairs/vets-website/pull/16125 is merged. I don't want him to lose track of the discussion that took place here.

**Comment by mnorthuis** [comment]
There is a larger conversation we need to revisit here regarding the H1s, the left nav and the breadcrumb.  That convo wasn't able to happen prior to Josh's time off.  I don't believe this is a launch blocker given Josh tagged it as a 2, and this is something that existed prior to the direct deposit work.  Please keep this ticket up until we can resolve the original issue. 

cc @1Copenut @noahgelman for awareness

**Comment by erikphansen** [comment]
@joshkimux ready to validate on staging.

**Comment by mshea0606** [comment]
@1Copenut could you validate this since @joshkimux is out?

**Comment by joshkimux** [comment]
@mshea0606 I'll validate tomorrow!

**Comment by joshkimux** [comment]
@erikphansen Just took some time to look through https://github.com/department-of-veterans-affairs/vets-website/pull/16125 ! I should have been more specific in this ticket, but the issue actually is not because of the _string_ "Direct deposit" itself, it's actually around its semantic heading level which is currently still a `h2`, not the `h1` as the page's `title` suggests.

The `h1` on this page is actually `Your Profile`, which is essentially a content/IA problem for _all_ of my profile that needs to be tackled first before a code fix can be made.

To summarize, we must either:
- change the `h1` to be "Direct Deposit," not "Your Profile" in order to match the page title "Direct Deposit" OR
- change the page `title` to be "Your Profile" to match the current `h1` "Your Profile"

I hope that clears it up a bit more! I can explain in more detail here or over a call 🙇

**Comment by joshkimux** [comment]
@tressaellen and @Samara-Strauss , by any chance were either of you involved in the design process for my profile? We're looking to find a solution to this ticket, but could use additional context and guidance from the original design team 🙇

**Comment by Samara-Strauss** [comment]
@tressaellen was the designer on the Profile redesign. She can chime in as time allows!

**Comment by tressaellen** [comment]
@joshkimux I'm happy to chat about this whenever works for you!

**Comment by joshkimux** [comment]
@tressaellen Awesome! I've set up a time for us to chat. @Samara-Strauss I've added you as optional to that chat as well, but feel free to skip!

**Comment by Samara-Strauss** [comment]
@joshkimux thanks! I responded over email as well, but cross-posting here — I'll sit this chat out because I don't think I'll be very helpful, but can you also include @ajakabcin and @andaleliz? They won't have the background knowledge Tressa has, but this will be a good orientation to some profile things for AJ (our new PM) and for Liz (new to auth exp and will primarily be focusing on the profile going forward).

**Comment by joshkimux** [comment]
@Samara-Strauss can do! Adding them on 😄

**Comment by Samara-Strauss** [comment]
Great, thank you!

**Comment by mnorthuis** [comment]
Here are the scribbles for the solution we discussed:
- The H1 would be moved from the sidenav title to the main heading of the content .
- The sidenav heading, would be an H4 (this is based on how this is implemented on the rest of the site, this is a separate issue, that will discuss separately and may impact this, but this is the working assumption for now)
- The title tag will pull the H1, as per standards
- The breadcrumb will need to be modified to exclude the "Your profile" segment of the URL.  Instead, "Profile:" will be appended to the beginning of each H1 in the current page segment of the breadcrumb. 
- No changes will be made to URLs

![image](https://user-images.githubusercontent.com/20994159/112209569-9cd76480-8be7-11eb-8530-f60a25d5efe9.png)

**Comment by Samara-Strauss** [comment]
Does this require any design work at this point, or is this something @erikphansen can work on pending availability?

**Comment by erikphansen** [comment]
Will this change apply to the mobile view as well? Note that currently the mobile nav button has a "Profile menu" label where "Profile" is wrapped in an `h1` 
![image](https://user-images.githubusercontent.com/20728956/112222945-7533cc80-8be6-11eb-9b1a-853c216e3761.png)

**Comment by joshkimux** [comment]
@erikphansen Yes, that's correct. This would be a `h4` to be consistent with other pages on VA.gov for now, but will need to be changed again in the future as a sitewide ticket to a normal `p` tag that is linked to the `nav` using `aria-labelledby`.

**Comment by andaleliz** [comment]
@Samara-Strauss to answer your question:
> Does this require any design work at this point

It shouldn't. Nothing about the design is changing other than the breadcrumb pattern, which I don't think @erikphansen needs represented in a mock-up given what's outlined above.  Erik, lmk if I'm wrong on that and you do need a mock-up

**Comment by Samara-Strauss** [comment]
Thanks @andaleliz! I can put this down for Erik for sprint 45.

**Comment by andaleliz** [comment]
@joshkimux and @mnorthuis, I think the mobile breadcrumb would now be `< Home` - can you confirm? 

It's currently `< Profile`

**Comment by joshkimux** [comment]
@andaleliz Correct, it would be `< Home`

**Comment by erikphansen** [comment]
[I just merged a this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/16879) to address this. So it can be validated on staging later today or tomorrow. Actually it'll probably get to prod first with today's daily deploy. 🙄

**Comment by ajakabcin** [comment]
I checked this on staging and everything looks good to me except one minor thing. @mnorthuis , I noticed in your [comment here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/19552#issuecomment-805196035), you said the url should be /**profile/contact-information** but I noticed on staging we have **/profile/personal-information**, however I also noticed your summary said "no changes will be made to URLs". I'm not sure if this was a just a typo in our notes or if we do in fact need to change the url. 

Staging:
![image](https://user-images.githubusercontent.com/45603961/115915390-44db8a00-a441-11eb-8d1f-a12fff7357d2.png)


I flagged this for @erikphansen (and @andaleliz) for discussion as well, and Erik made a good point that if we do need to change this url we'll need a redirect and updates to links since there are likely other places on the site that link to this section.

cc @Samara-Strauss and @joshkimux

**Comment by Samara-Strauss** [comment]
We should definitely wait to hear back from @mnorthuis before closing anything out, but if there is follow-up work to modify or redirect a URL, we should create a separate ticket for that and close this out since it sounds like the H1 issue is resolved.

**Comment by mnorthuis** [comment]
Hi, no URL changes are needed, my documentation was out of date, so I will update it to reflect the current URLs in production.  Sorry for the confusion!

**Comment by joshkimux** [comment]
@Samara-Strauss I can give a thumbs up from accessibility that this is looking good. Great work to the team for this huge a11y improvement 🥳 

Important note, this change has introduced the need for an additional fix. Since the former `h2`s are now `h1`s, all the section headings need to be bumped up by one from `h3`s to `h2`s.

[I've created a new ticket to address heading level fixes.](https://github.com/department-of-veterans-affairs/va.gov-team/issues/23756)

**Comment by Samara-Strauss** [comment]
@mnorthuis sounds good! Thank you for clarifying.

@joshkimux got it, and thank you for creating that ticket for us. cc/ @ajakabcin for awareness that there's a follow-up ticket coming out of this work.

Closing this ticket since it seems like the original requirements have all been met.
